### PR TITLE
tools/ramdump : Show heap usage in ramdump

### PR DESCRIPTION
**Pull request by YiHyunJin**
Details of Heap Usages (Size in Bytes)

******************************************************************
  MemAddr |   Size   | Status |  Pid  |           Owner
----------|----------|--------|-------|---------------------------
0x2025ba0 |    528   |  Alloc |   0   | os/kernel/signal/sig_initialize.c:170
0x2025db0 |    272   |  Alloc |   0   | os/kernel/signal/sig_initialize.c:175
0x2025ec0 |    336   |  Alloc |   0   | os/kernel/signal/sig_initialize.c:260
0x2026010 |    400   |  Alloc |   0   | os/kernel/signal/sig_initialize.c:203
0x20261a0 |    208   |  Alloc |   0   | os/kernel/signal/sig_initialize.c:203
0x2026270 |   2448   |  Alloc |   0   | os/kernel/mqueue/mq_initialize.c:149
0x2026c00 |   4880   |  Alloc |   0   | os/kernel/mqueue/mq_initialize.c:149
0x2027f10 |    320   |  Alloc |   0   | os/kernel/mqueue/mq_initialize.c:227
0x2028050 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x2028080 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x20280b0 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x20280e0 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x2028110 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x2028140 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x2028170 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x20281a0 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x20281d0 |   1760   |  Alloc |   0   | os/kernel/group/group_create.c:201
0x20288b0 |     80   |  Alloc |   0   | os/fs/vfs/fs_fdopen.c:242
0x2028900 |     80   |  Alloc |   0   | os/fs/vfs/fs_fdopen.c:242
0x2028950 |     80   |  Alloc |   0   | os/fs/vfs/fs_fdopen.c:242
0x20289a0 |    256   |  Alloc |   0   | os/kernel/task/task_create.c:149
0x2028aa0 |   1760   |  Alloc |   0   | os/kernel/group/group_create.c:201
0x2029180 |     80   |  Alloc |   0   | os/fs/vfs/fs_fdopen.c:242
0x20291d0 |     80   |  Alloc |   0   | os/fs/vfs/fs_fdopen.c:242
0x2029220 |     80   |  Alloc |   0   | os/fs/vfs/fs_fdopen.c:242
0x2029270 |   2064   |  Alloc |   1   | os/arch/arm/src/common/up_createstack.c:188
0x2029a80 |     96   |  Alloc |   0   | os/fs/driver/mtd/mtd_partition.c:740
0x2029ae0 |     96   |  Alloc |   0   | os/fs/driver/mtd/mtd_partition.c:740
0x2029b40 |     48   |  Alloc |   0   | os/fs/driver/mtd/ftl.c:512
0x2029b70 |   4112   |  Alloc |   0   | os/fs/driver/mtd/ftl.c:532
0x202ab80 |     64   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202abc0 |     96   |  Alloc |   0   | os/fs/driver/mtd/mtd_partition.c:740
0x202ac20 |     96   |  Alloc |   0   | os/fs/driver/mtd/mtd_partition.c:740
0x202ac80 |     96   |  Alloc |   0   | os/fs/driver/mtd/mtd_partition.c:740
0x202ace0 |     96   |  Alloc |   0   | os/fs/driver/mtd/mtd_partition.c:740
0x202ad40 |     96   |  Alloc |   0   | os/fs/driver/mtd/mtd_partition.c:740
0x202ada0 |     96   |  Alloc |   0   | os/fs/driver/mtd/mtd_partition.c:740
0x202ae00 |     48   |  Alloc |   0   | os/fs/driver/mtd/ftl.c:512
0x202ae30 |   4112   |  Alloc |   0   | os/fs/driver/mtd/ftl.c:532
0x202be40 |     64   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202be80 |     96   |  Alloc |   0   | os/fs/driver/mtd/mtd_partition.c:740
0x202bee0 |     96   |  Alloc |   0   | os/fs/driver/mtd/smart.c:5208
0x202bf40 |   6320   |  Alloc |   0   | os/fs/driver/mtd/smart.c:1086
0x202d7f0 |    528   |  Alloc |   0   | os/fs/driver/mtd/smart.c:1175
0x202da00 |    528   |  Alloc |   0   | os/fs/driver/mtd/smart.c:1181
0x202dc10 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202dc40 |     96   |  Alloc |   0   | os/fs/driver/mtd/mtd_partition.c:740
0x202dca0 |     48   |  Alloc |   0   | os/fs/driver/mtd/mtd_config.c:1225
0x202dcd0 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202dd00 |     96   |  Alloc |   0   | os/fs/driver/mtd/mtd_partition.c:740
0x202dd60 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202dd90 |     64   |  Alloc |   0   | os/fs/smartfs/smartfs_smart.c:1545
0x202ddd0 |    528   |  Alloc |   0   | os/fs/smartfs/smartfs_utils.c:603
0x202dfe0 |    272   |  Alloc |   0   | os/fs/smartfs/smartfs_utils.c:605
0x202e0f0 |     32   |  Alloc |   0   | os/drivers/rtc.c:619
0x202e110 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202e140 |     32   |  Alloc |   0   | os/arch/arm/src/chip/s5j_gpio_lowerhalf.c:256
0x202e160 |     32   |  Alloc |   0   | os/drivers/gpio/gpio.c:691
0x202e180 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202e1b0 |     32   |  Alloc |   0   | os/arch/arm/src/chip/s5j_gpio_lowerhalf.c:256
0x202e1d0 |     32   |  Alloc |   0   | os/drivers/gpio/gpio.c:691
0x202e1f0 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202e220 |     32   |  Alloc |   0   | os/arch/arm/src/chip/s5j_gpio_lowerhalf.c:256
0x202e240 |     32   |  Alloc |   0   | os/drivers/gpio/gpio.c:691
0x202e260 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202e290 |     32   |  Alloc |   0   | os/arch/arm/src/chip/s5j_gpio_lowerhalf.c:256
0x202e2b0 |     32   |  Alloc |   0   | os/drivers/gpio/gpio.c:691
0x202e2d0 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202e300 |     32   |  Alloc |   0   | os/arch/arm/src/chip/s5j_gpio_lowerhalf.c:256
0x202e320 |     32   |  Alloc |   0   | os/drivers/gpio/gpio.c:691
0x202e340 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202e370 |     32   |  Alloc |   0   | os/arch/arm/src/chip/s5j_gpio_lowerhalf.c:256
0x202e390 |     32   |  Alloc |   0   | os/drivers/gpio/gpio.c:691
0x202e3b0 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202e3e0 |     32   |  Alloc |   0   | os/arch/arm/src/chip/s5j_gpio_lowerhalf.c:256
0x202e400 |     32   |  Alloc |   0   | os/drivers/gpio/gpio.c:691
0x202e420 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202e450 |     32   |  Alloc |   0   | os/arch/arm/src/chip/s5j_gpio_lowerhalf.c:256
0x202e470 |     32   |  Alloc |   0   | os/drivers/gpio/gpio.c:691
0x202e490 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202e4c0 |     32   |  Alloc |   0   | os/arch/arm/src/chip/s5j_gpio_lowerhalf.c:256
0x202e4e0 |     32   |  Alloc |   0   | os/drivers/gpio/gpio.c:691
0x202e500 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202e530 |     32   |  Alloc |   0   | os/arch/arm/src/chip/s5j_gpio_lowerhalf.c:256
0x202e550 |     32   |  Alloc |   0   | os/drivers/gpio/gpio.c:691
0x202e570 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202e5a0 |     32   |  Alloc |   0   | os/arch/arm/src/chip/s5j_gpio_lowerhalf.c:256
0x202e5c0 |     32   |  Alloc |   0   | os/drivers/gpio/gpio.c:691
0x202e5e0 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202e610 |     32   |  Alloc |   0   | os/arch/arm/src/chip/s5j_gpio_lowerhalf.c:256
0x202e630 |     32   |  Alloc |   0   | os/drivers/gpio/gpio.c:691
0x202e650 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202e680 |     32   |  Alloc |   0   | os/arch/arm/src/chip/s5j_gpio_lowerhalf.c:256
0x202e6a0 |     32   |  Alloc |   0   | os/drivers/gpio/gpio.c:691
0x202e6c0 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202e6f0 |     32   |  Alloc |   0   | os/arch/arm/src/chip/s5j_gpio_lowerhalf.c:256
0x202e710 |     32   |  Alloc |   0   | os/drivers/gpio/gpio.c:691
0x202e730 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202e760 |     32   |  Alloc |   0   | os/arch/arm/src/chip/s5j_gpio_lowerhalf.c:256
0x202e780 |     32   |  Alloc |   0   | os/drivers/gpio/gpio.c:691
0x202e7a0 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202e7d0 |     32   |  Alloc |   0   | os/arch/arm/src/chip/s5j_gpio_lowerhalf.c:256
0x202e7f0 |     32   |  Alloc |   0   | os/drivers/gpio/gpio.c:691
0x202e810 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202e840 |     32   |  Alloc |   0   | os/arch/arm/src/chip/s5j_gpio_lowerhalf.c:256
0x202e860 |     32   |  Alloc |   0   | os/drivers/gpio/gpio.c:691
0x202e880 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202e8b0 |     32   |  Alloc |   0   | os/arch/arm/src/chip/s5j_gpio_lowerhalf.c:256
0x202e8d0 |     32   |  Alloc |   0   | os/drivers/gpio/gpio.c:691
0x202e8f0 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202e920 |     32   |  Alloc |   0   | os/arch/arm/src/chip/s5j_gpio_lowerhalf.c:256
0x202e940 |     32   |  Alloc |   0   | os/drivers/gpio/gpio.c:691
0x202e960 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202e990 |     32   |  Alloc |   0   | os/arch/arm/src/chip/s5j_gpio_lowerhalf.c:256
0x202e9b0 |     32   |  Alloc |   0   | os/drivers/gpio/gpio.c:691
0x202e9d0 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202ea00 |     32   |  Alloc |   0   | os/arch/arm/src/chip/s5j_gpio_lowerhalf.c:256
0x202ea20 |     32   |  Alloc |   0   | os/drivers/gpio/gpio.c:691
0x202ea40 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202ea70 |     32   |  Alloc |   0   | os/arch/arm/src/chip/s5j_gpio_lowerhalf.c:256
0x202ea90 |     32   |  Alloc |   0   | os/drivers/gpio/gpio.c:691
0x202eab0 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202eae0 |     32   |  Alloc |   0   | os/arch/arm/src/chip/s5j_gpio_lowerhalf.c:256
0x202eb00 |     32   |  Alloc |   0   | os/drivers/gpio/gpio.c:691
0x202eb20 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202eb50 |     32   |  Alloc |   0   | os/arch/arm/src/chip/s5j_gpio_lowerhalf.c:256
0x202eb70 |     32   |  Alloc |   0   | os/drivers/gpio/gpio.c:691
0x202eb90 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202ebc0 |     32   |  Alloc |   0   | os/arch/arm/src/chip/s5j_gpio_lowerhalf.c:256
0x202ebe0 |     32   |  Alloc |   0   | os/drivers/gpio/gpio.c:691
0x202ec00 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202ec30 |     32   |  Alloc |   0   | os/arch/arm/src/chip/s5j_gpio_lowerhalf.c:256
0x202ec50 |     32   |  Alloc |   0   | os/drivers/gpio/gpio.c:691
0x202ec70 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202eca0 |     32   |  Alloc |   0   | os/arch/arm/src/chip/s5j_gpio_lowerhalf.c:256
0x202ecc0 |     32   |  Alloc |   0   | os/drivers/gpio/gpio.c:691
0x202ece0 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202ed10 |     32   |  Alloc |   0   | os/arch/arm/src/chip/s5j_gpio_lowerhalf.c:256
0x202ed30 |     32   |  Alloc |   0   | os/drivers/gpio/gpio.c:691
0x202ed50 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202ed80 |     32   |  Alloc |   0   | os/arch/arm/src/chip/s5j_gpio_lowerhalf.c:256
0x202eda0 |     32   |  Alloc |   0   | os/drivers/gpio/gpio.c:691
0x202edc0 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202edf0 |     32   |  Alloc |   0   | os/arch/arm/src/chip/s5j_gpio_lowerhalf.c:256
0x202ee10 |     32   |  Alloc |   0   | os/drivers/gpio/gpio.c:691
0x202ee30 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202ee60 |     32   |  Alloc |   0   | os/arch/arm/src/chip/s5j_gpio_lowerhalf.c:256
0x202ee80 |     32   |  Alloc |   0   | os/drivers/gpio/gpio.c:691
0x202eea0 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202eed0 |     32   |  Alloc |   0   | os/arch/arm/src/chip/s5j_gpio_lowerhalf.c:256
0x202eef0 |     32   |  Alloc |   0   | os/drivers/gpio/gpio.c:691
0x202ef10 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202ef40 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202ef70 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202efa0 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202efd0 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202f000 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202f030 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202f060 |     48   |  Alloc |   0   | os/fs/inode/fs_inodereserve.c:117
0x202f090 |    256   |  Alloc |   0   | os/kernel/task/task_create.c:149
0x202f190 |   1760   |  Alloc |   0   | os/kernel/group/group_create.c:201
0x202f870 |     80   |  Alloc |   0   | os/fs/vfs/fs_fdopen.c:242
0x202f8c0 |     80   |  Alloc |   0   | os/fs/vfs/fs_fdopen.c:242
0x202f910 |     80   |  Alloc |   0   | os/fs/vfs/fs_fdopen.c:242
0x202f960 |   1040   |  Alloc |   4   | apps/examples/hello/hello_main.c:76
0x202fd70 |    272   |  Alloc |   4   | apps/examples/hello/hello_main.c:77
0x202fe80 |     48   |  Alloc |   4   | apps/examples/hello/hello_main.c:78
0x202feb0 |    656   |  Free  |       |
0x2030140 |    256   |  Alloc |   2   | os/kernel/task/task_create.c:149
0x2030240 |   1760   |  Alloc |   2   | os/kernel/group/group_create.c:201
0x2030920 |     80   |  Alloc |   2   | os/fs/vfs/fs_fdopen.c:242
0x2030970 |     80   |  Alloc |   2   | os/fs/vfs/fs_fdopen.c:242
0x20309c0 |     80   |  Alloc |   2   | os/fs/vfs/fs_fdopen.c:242
0x2030a10 |   4112   |  Alloc |   3   | os/arch/arm/src/common/up_createstack.c:188
0x2031a20 |    144   |  Alloc |   3   | apps/shell/tash_main.c:92
0x2031ab0 |     48   |  Alloc |   3   | os/fs/vfs/fs_select.c:151
0x2031ae0 |   2064   |  Alloc |   4   | os/arch/arm/src/common/up_createstack.c:188
0x20322f0 | 908544   |  Free  |       |

**********************************************
   Summary of Heap Usages (Size in Bytes)
**********************************************
HEAP SIZE        :  959600
PEAK ALLOC SIZE  :  50368
TOTAL ALLOC SIZE :  50368
FREE SIZE        :  909232

***********************************************
  PID  |   CUR ALLOC SIZE   | PEAK ALLOC SIZE |
-------|--------------------|-----------------|
   0   |          38320     |         40384   |
   1   |              0     |             0   |
   3   |            192     |           192   |
   4   |           1360     |          1360   |

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [c3d39726e33b792a2fda840c81466b517a5cc18c] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [c9024eb6ac8e5c1ee7754e32d52c66bb73b70d1d] - Code Rule Check OK.

File: tools/ramdump/dumpParser.py [hunk 1]
```diff
@@ -702,6 +702,109 @@ def main():
...
 
+		print ''
+		print ''
+		print 'Details of Heap Usages (Size in Bytes)'
+		print ''
+
+		g_mmheap = rParser.get_address_of_symbol("g_mmheap")
+
+		# Read config informaiton
+		fd = open("../../os/.config", 'r')
```
Comment (jeongchanKim) [inline tools/ramdump/dumpParser.py]: How about defining PATH for many cases?
If folder structure changed, we can modify the path easily.

File: tools/ramdump/dumpParser.py [hunk 2]
```diff
@@ -702,6 +702,109 @@ def main():
...
+		data = fd.read()
+		fd.close()
+
+		MM_SEM_SIZE = 4
+		MM_HOLDER_SIZE = 4
+		MM_COUNTS_HELD_SIZE = 4
+		MM_HEAPSIZE_SIZE = 4
+
+		HEPINFO_TCB_SIZE = 16
```
Comment (jeongchanKim) [inline tools/ramdump/dumpParser.py]: HEPINFO -> HEAPINFO

File: tools/ramdump/dumpParser.py [hunk 3]
```diff
@@ -702,6 +702,109 @@ def main():
...
+		g_mmheap = rParser.get_address_of_symbol("g_mmheap")
+
+		# Read config informaiton
+		fd = open("../../os/.config", 'r')
+		data = fd.read()
+		fd.close()
+
+		MM_SEM_SIZE = 4
+		MM_HOLDER_SIZE = 4
+		MM_COUNTS_HELD_SIZE = 4
```
Comment (jeongchanKim) [inline tools/ramdump/dumpParser.py]: What is the meaning of HELD_SIZE?

File: tools/ramdump/dumpParser.py [hunk 4]
```diff
@@ -702,6 +702,109 @@ def main():
...
+			while data[index] != '\n' :
+				max_task *= 10
+				max_task += int(data[index])
+				index += 1
+
+		if 'CONFIG_DEBUG_MM_HEAPINFO=y' in data:
+			START_HEAP_POINT += 8
+			START_HEAP_POINT += (max_task * HEPINFO_TCB_SIZE)
+		else :
+			print 'DEBUG_MM_HEAPINFO is not enable. Enalbe DEBUG_MM_HEAPINFO to see heap usage'
```
Comment (jeongchanKim) [inline tools/ramdump/dumpParser.py]: Enalbe -> Enable

File: tools/ramdump/dumpParser.py [hunk 5]
```diff
@@ -702,6 +702,109 @@ def main():
...
+		print "  MemAddr |   Size   | Status |  Pid  |           Owner           "
+		print "----------|----------|--------|-------|---------------------------"
+		while point < end_heap:
+			size = rParser.read_word(point)
+			preceding = rParser.read_word(point + 4)
+			owner = rParser.read_word(point + 8)
+			pid = rParser.read_halfword(point + 12)
+			if preceding & MM_ALLOC_BIT :
+				if pid >= 0:
+					cmd = ['addr2line', '-e', '../../build/output/bin/tinyara', hex(owner)]
```
Comment (jeongchanKim) [inline tools/ramdump/dumpParser.py]: We can declare cmd in the above for common.

File: tools/ramdump/dumpParser.py [hunk 6]
```diff
@@ -702,6 +702,109 @@ def main():
...
+		heap_size = rParser.read_word(g_mmheap + 12)
+		print 'HEAP SIZE        : ', heap_size
+		print 'PEAK ALLOC SIZE  : ', rParser.read_word(g_mmheap + 16)
+
+		total_alloc_size = rParser.read_word(g_mmheap + 20)
+		print 'TOTAL ALLOC SIZE : ', total_alloc_size
+		print 'FREE SIZE        : ', heap_size - total_alloc_size
+		print ''
+		print "***********************************************"
+		print "  PID  |   CUR ALLOC SIZE   | PEAK ALLOC SIZE |"
```
Comment (jeongchanKim) [inline tools/ramdump/dumpParser.py]: How about showing task's stack size?

File: tools/ramdump/dumpParser.py [hunk 7]
```diff
@@ -702,6 +702,109 @@ def main():
 		print '&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&'
 
 
+		print ''
+		print ''
```
Comment (jeongchanKim) [inline tools/ramdump/dumpParser.py]: How about showing the start and end address for heap?

File: tools/ramdump/dumpParser.py [hunk 8]
```diff
@@ -702,6 +702,109 @@ def main():
...
+		print 'Details of Heap Usages (Size in Bytes)'
+		print ''
+
+		g_mmheap = rParser.get_address_of_symbol("g_mmheap")
+
+		# Read config informaiton
```
Comment (jeongchanKim) [inline tools/ramdump/dumpParser.py]: informaiton -> information

File: tools/ramdump/dumpParser.py [hunk 9]
```diff
@@ -702,6 +702,109 @@ def main():
...
+		INVALID_PROCESS_ID = 0xFFFFFFFF
+
+		alloc_list = g_mmheap + 24
+		for i in range(0, max_task):
+			pid = rParser.read_word(alloc_list)
+			if pid != INVALID_PROCESS_ID :
+				cur_alloc = rParser.read_word(alloc_list + 4)
+				peak_alloc = rParser.read_word(alloc_list + 8)
+				print '{:^7}|'.format(pid), "{:>14}".format(cur_alloc), '    |', '{:>13}'.format(peak_alloc), '  |'
+			# next alooc list
```
Comment (jeongchanKim) [inline tools/ramdump/dumpParser.py]: alooc -> alloc

**Comment by seinfra** [comment]
Target : [188e6bbb0f24057fd4a86952b4e40ebaeb35162d] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [188e6bbb0f24057fd4a86952b4e40ebaeb35162d] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [8b32c7016ea2966b422333e8c7f5cc8c8c12c43f] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [8b32c7016ea2966b422333e8c7f5cc8c8c12c43f] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [703c00e813cec68b11c25eb6e2a3dffa43ced864] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [703c00e813cec68b11c25eb6e2a3dffa43ced864] - Code Rule Check (C++) OK.

File: tools/ramdump/dumpParser.py [hunk 10]
```diff
@@ -37,6 +37,10 @@
 g_stext=0
 g_etext=0
 
+config_path = '../../os/.config'
+elf_path = '../../build/output/bin/tinyara'
+debug_cmm = 'addr2line'
```
Comment (jeongchanKim) [inline tools/ramdump/dumpParser.py]: usually, we use cmd for command

**Comment by seinfra** [comment]
Target : [d4478dfe43ca2bc7f3cf47ab4096b4507b380dfe] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [d4478dfe43ca2bc7f3cf47ab4096b4507b380dfe] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [c21ed9dd02f31e41bce5014c3ae940a664ededf2] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [c21ed9dd02f31e41bce5014c3ae940a664ededf2] - Code Rule Check OK.

File: tools/ramdump/dumpParser.py [hunk 11]
```diff
@@ -702,6 +706,126 @@ def main():
...
+				data = fd_popen.read()
+				fd_popen.close()
+				if pid >= 0:
+					print '{:^10}|'.format(hex(point)), '{:>6}'.format(size), '  |', '{:^7}|'.format('Alloc'), '{:^6}|'.format(pid), data[14:],
+				else: # If pid is less than 0, it is in the (-pid)'s stack size
+					stack_size[(-pid) & (max_tasks - 1)] = size
+					print '{:^10}|'.format(hex(point)), '{:>6}'.format(size), '  |', '{:^7}|'.format('Alloc'), '{:^6}|'.format(-pid), data[14:],
+			else:
+				print '{:^10}|'.format(hex(point)), '{:>6}'.format(size), '  |', '{:^7}|'.format('Free'), '{:6}|'.format("")
+			# next alloc node
```
Comment (jeongchanKim) [inline tools/ramdump/dumpParser.py]: next is not always alloc node. Let's change to node.

**Comment by seinfra** [comment]
Target : [23d939f7dcbabf5833664bfb1805045586454d5b] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [23d939f7dcbabf5833664bfb1805045586454d5b] - Code Rule Check (C++) OK.

**Pull request merged**